pkg/admission/types.go

```diff
@@ -31,3 +32,9 @@ type WantsInformerFactory interface {
 	SetInformerFactory(informers.SharedInformerFactory)
 	Validator
 }
+
+// WantsAuthorizer defines a function which sets Authorizer for admission plugins that need it.
+type WantsAuthorizer interface {
+	SetAuthorizer(authorizer.Authorizer)
+	Validator
```

Comment:
> @deads2k @krousey - any more thoughts here? In either case I need to get an authorizer into an admission plugin, whether that infrastructure is a config object all pieces get or the Wants pattern. Thanks for the reviews so far.

I really like prefer having the `Validator` interface required for all the `Wants*` types, but @krousey seems to feel pretty strongly.  I'd give it up to get the rest of the pull.